### PR TITLE
[stable/spinnaker] Adds support for pre-created secrets and removes ingress rules when no host is supplied

### LINKED ISSUE #9731: [stable/spinnaker] use pre-created secrets for all credentials
**Is this a request for help?**: No

---

**FEATURE REQUEST**

**Version of Helm and Kubernetes**:
```
Client: &version.Version{SemVer:"v2.8.0", GitCommit:"14af25f1de6832228539259b821949d20069a222", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.0", GitCommit:"14af25f1de6832228539259b821949d20069a222", GitTreeState:"clean"}
```

**Which chart**:
`stable/spinnaker`

**What happened**:
On the Spinnaker chart, you can make reference to an already set secret for the Docker Registry, but you have to change the values file to include the credentials for GCS, S3, etc. 

**What you expected to happen**:
It would be great if one could make reference to secrets instead of hardcoding them on the values file, as already happens to the Docker Registry credentials. 

**How to reproduce it** (as minimally and precisely as possible):
On the chart, you are allowed to do this
`# dockerRegistryAccountSecret: myregistry-secrets`

So you can create a secret from a file and reference it
`kubectl create secret generic gcr-secrets -n spinnaker --from-file=gcr="credentials.json"`

For GCS, you have to copy and paste a b64 encoded credentials file

```
gcs:
  enabled: false
  project: my-project-name
  bucket: "<GCS-BUCKET-NAME>"
  jsonKey: '<INSERT CLOUD STORAGE JSON HERE>'
```

### PR DESCRIPTION
**Pull request by retpolanne**
#### What this PR does / why we need it:
It allows you to use pre-created secrets for S3 and GCS instead of hardcoding them on the `values.yaml` and supplying them to Tiller. We need it because we don't want to keep secrets hardcoded and supply secrets to Tiller. 

It also creates ingress without rules when no host is supplied. We need it because, if you are using a setting where your ingress should be a catch-all (such as when using a GCE static ip), you shouldn't supply any rules to the ingress. 

It also allows to use any Docker registry on Halyard's kubernetes account (dockerhub was hardcoded).

#### Which issue this PR fixes
  - fixes #9731, #9822

#### Special notes for your reviewer:
I kept comments on how the variables I added work, but I didn't explain it in depth. It might be good, maybe on another PR, to update the README and to explain the variables. 

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

**Review by dwardu89 — APPROVED** [review]
LGTM

File: stable/spinnaker/templates/statefulsets/halyard.yaml [hunk 1]
```diff
@@ -66,7 +70,11 @@ spec:
       {{- if and .Values.s3.enabled .Values.s3.accessKey .Values.s3.secretKey }}
```
Comment (vyasswaroop) [inline stable/spinnaker/templates/statefulsets/halyard.yaml:69]: Here also you'd have to include .Values.s3.secretName

**Review by vyasswaroop — COMMENTED** [review]
Also would have to include the value of s3 secretName to the following lines as well. https://github.com/helm/charts/blob/75acd8f310870e7faebe2d87d322399bdbbbf140/stable/spinnaker/templates/configmap/halyard-config.yaml#L42

File: stable/spinnaker/templates/statefulsets/halyard.yaml (hunk 1 above)
Comment (retpolanne) [reply stable/spinnaker/templates/statefulsets/halyard.yaml:69]: @vyasswaroop forgot to put it [here](https://github.com/helm/charts/blob/b3c167d9f92bad05118ecfce865d0f0a238b7914/stable/spinnaker/templates/configmap/halyard-config.yaml#L53) too, thanks.

**Pull request closed without merging**